wallet/src/de/schildbach/wallet/ui/more/TransactionMetadataSettingsViewModel.kt

```
@@ -0,0 +1,120 @@
+
+    val saveToNetwork = dashPayConfig.observe(DashPayConfig.TRANSACTION_METADATA_SAVE_TO_NETWORK)
+
+    fun getBalanceInLocalFormat(): String {
+        selectedExchangeRate.value?.fiat?.let {
+            val exchangeRate = org.bitcoinj.utils.ExchangeRate(Coin.COIN, it)
+            val fiatValue = exchangeRate.coinToFiat(CURRENT_DATA_COST)
+            val minValue = try {
+                val fractionDigits = Currency.getInstance(selectedCurrency).defaultFractionDigits
+                val newValue = BigDecimal.ONE.movePointLeft(fractionDigits)
+                Fiat.parseFiat(fiatValue.currencyCode, newValue.toPlainString())
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)